ext/case_transform/src/lib.rs

```diff
@@ -2,80 +2,128 @@
 extern crate ruru;
 extern crate inflector;
 
-// // dash: kebab-case
-use inflector::cases::kebabcase::to_kebab_case;
-// // underscore: snake_case
-use inflector::cases::snakecase::to_snake_case;
-// // camel_lower: camelCase
-use inflector::cases::camelcase::to_camel_case;
-// // camel: ClassCase (PascalCase)
-use inflector::cases::classcase::to_class_case;
-
-use ruru::{Class, Object, RString, Hash, Array, Symbol, AnyObject, VM};
+use inflector::cases::{camelcase, classcase, kebabcase, snakecase};
+
+use ruru::{Class, Object, VerifiedObject, RString, Hash, Array, Symbol, AnyObject};
 use ruru::types::ValueType;
+use ruru::result::Error as RuruError;
 
-class!(CaseTransform);
+trait Transform: Object {
```

Comment:
`Transform` describes an interface for transformation. It can be implemented only to Ruby objects due to `Object` constraint.

Then we make type-specific implementations for

 - `AnyObject` (no-op)
 - `RString`
 - `Hash`
 - `Array`

by implementing `transorm` methods.